OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -82,6 +95,11 @@ public sealed class HackyAIInfo : IBotInfo, ITraitInfo
 		[Desc("Radius in cells around the center of the base to expand.")]
 		public readonly int MaxBaseRadius = 20;
 
+		[Desc("Radius in cells around each building with ProvideBuildableArea",
+		"to check for a 3x3 area of water where naval structures can be built.",
```

Comment:
Factoring in bulding footprints (or generally more than 3x3 cells) would need another refactor, and I don't feel like doing that in this pr. This is already far better than what we had before.